pubsublite/pscompat/publisher.go

```diff
@@ -99,7 +106,15 @@ func NewPublisherClientWithSettings(ctx context.Context, topic string, settings
 	if err := wirePub.WaitStarted(); err != nil {
 		return nil, err
 	}
-	return &PublisherClient{settings: settings, wirePub: wirePub}, nil
+	publisher := &PublisherClient{settings: settings, wirePub: wirePub}
+
+	// Mitigation for Stop not being called when the publisher client is no longer
+	// used. Users must still call Stop to promptly shut down the publisher, as
+	// finalizers run after an arbitrary amount of time.
+	runtime.SetFinalizer(publisher, func(p *PublisherClient) {
+		p.wirePub.Stop()
```

Comment:
I am not sure I would do it now, but it might be good to drop a TODO here about adding a log warning eventually. In the not too distant future our libraries will have better access to a logger that can be used and configured by users. It would eventually be nice to be able to nudge users in the write direction and call Stop

Reply:
+1

Reply:
Glad to hear that a logger will be available soon!
Added a TODO.